sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/GroupServiceImpl.java

```diff
@@ -77,11 +79,38 @@ public Group findById(String realm, String appName, String id) {
 
   @Override
   public void update(String realm, String appName, Group group) {
+    group.setUsers(new ArrayList<>());
     sugoiEventPublisher.publishCustomEvent(
         realm,
         null,
         SugoiEventTypeEnum.UPDATE_GROUP,
         Map.ofEntries(Map.entry("group", group), Map.entry("appName", appName)));
     storeProvider.getWriterStore(realm).updateGroup(appName, group);
   }
+
+  @Override
+  public void addUserToGroup(String realm, String userId, String appName, String groupName) {
+    sugoiEventPublisher.publishCustomEvent(
+        realm,
+        null,
+        SugoiEventTypeEnum.ADD_USER_TO_GROUP,
+        Map.ofEntries(
+            Map.entry("user", userId),
+            Map.entry("appName", appName),
+            Map.entry("groupName", groupName)));
+    storeProvider.getWriterStore(realm).addUserToGroup(appName, groupName, userId);
```

Comment:
We might need to change provider interface here to add a `User` instead of the String userId, to verify that the user exists (before trying to add it), and maybe get the good storage (especially for ldap provider)

Reply:
Unstead of doing getWriterStore(realm) we could do a getWriterStore(realm,userstorage) where the userstorage come from findbyId(userId). what do you think of this ?

Reply:
sounds good, with error checking of course :)